web/src/app/urql.ts

```diff
@@ -0,0 +1,92 @@
+import { retryExchange } from '@urql/exchange-retry'
+import {
+  cacheExchange,
+  createClient,
+  dedupExchange,
+  fetchExchange,
+  Exchange,
+  Operation,
+} from 'urql'
+import { pipe, tap } from 'wonka'
+import { pathPrefix } from './env'
+
+const refetch: Array<(force: boolean) => void> = []
+export function refetchAll(force = false): void {
+  refetch.forEach((refetch) => refetch(force))
+}
+
+// allow refetching all active queries at any time
+const refetchExchange = (): Exchange => {
+  return ({ client, forward }) =>
+    (ops) => {
+      const watchedOps$ = new Map<number, Operation>()
+      const observedOps$ = new Map<number, number>()
+
+      refetch.push((force) => {
+        if (!force) {
+          // use existing policy (cache-first will not re-fetch)
+          watchedOps$.forEach((op) => {
+            client.reexecuteOperation(
+              client.createRequestOperation('query', op),
+            )
+          })
+          return
+        }
+
+        watchedOps$.forEach((op) => {
+          client.reexecuteOperation(
+            client.createRequestOperation('query', op, {
+              ...op.context,
+              requestPolicy: 'cache-and-network',
+            }),
+          )
+        })
+      })
+
+      const handleOp = (op: Operation): void => {
+        if (op.kind === 'query' && !observedOps$.has(op.key)) {
+          observedOps$.set(op.key, 1)
+          watchedOps$.set(op.key, op)
+        }
+
+        if (op.kind === 'teardown' && observedOps$.has(op.key)) {
+          observedOps$.delete(op.key)
+          watchedOps$.delete(op.key)
+        }
+      }
+
+      return forward(pipe(ops, tap(handleOp)))
+    }
+}
+
+// refetch every 15 sec or on refocus
+let poll: NodeJS.Timer
+function resetPoll(): void {
```

Comment:
maybe `setPoll` since it also includes setting it up the first time it executes?

```suggestion
function setPoll(): void {
```

Reply:
it clears the existing interval out to reset the timer and then creates a new one -- I don't think "set" would really cover that